SofaKernel/modules/SofaBaseMechanics/IdentityMapping.inl

```diff
@@ -75,6 +75,11 @@ void IdentityMapping<TIn, TOut>::apply(const core::MechanicalParams * /*mparams*
     helper::WriteOnlyAccessor< Data<VecCoord> > out = dOut;
     helper::ReadAccessor< Data<InVecCoord> > in = dIn;
 
+    if(out.size() != in.size()) // true if topological changes
+    {
```

Comment:
This indeed has nothing to do with SofaPython. It was added in cf5a31a5 "FIX out of bounds access in apply() with TopologicalModifiers scenes", I guess related to trying to fix crashes in tests or sample scenes. But I agree this is not the right way to fix this kind of issues.